misc/mzutil/scripts/wait_for_view_states.py

```diff
@@ -40,17 +41,43 @@ def view_names(
             yield row[0]
 
 
-def view_matches(cursor: psycopg2.extensions.cursor, view: str, expected: str) -> bool:
+def view_matches(
+    cursor: psycopg2.extensions.cursor, view: str, expected: str, timestamp: int
+) -> bool:
     """Return True if a SELECT from the VIEW matches the expected string."""
     stream = io.StringIO()
+    query = f"COPY (SELECT * FROM {view} WHERE mz_logical_timestamp() > {timestamp}) TO STDOUT"
     try:
-        cursor.copy_expert(f"COPY (SELECT * FROM {view}) TO STDOUT", stream)
+        cursor.copy_expert(query, stream)
     except psycopg2.errors.InternalError_:
         # The view is not yet ready to be queried
         return False
     return stream.getvalue() == expected
 
 
+def source_at_offset(
+    cursor: psycopg2.extensions.cursor, source_name: str, desired_offset: int
+) -> typing.Union[None, int]:
+    """Return the mz timestamp from a source if it has reached the desired offset."""
+    query = (
+        "SELECT timestamp FROM mz_source_info WHERE source_name = %s and offset = %s"
+    )
+    try:
+        cursor.execute(query, (source_name, desired_offset))
+        if cursor.rowcount > 1:
+            print("ERROR: More than one row returned when query source offsets:")
+            for row in cursor:
+                print(f"\t{row}")
+            sys.exit(1)
+        if not cursor.rowcount:
+            return None
+
+        return int(cursor.fetchone()[0])
+    except psycopg2.errors.InternalError_:
+        # The view is not yet ready to be queried
+        return False
```

Comment:
Should this return `None` instead of `False` ? (Maybe I'm missing something, because I wouldn't have expected this to type-check).

Reply:
You are right, this should be `None`. Not sure why it passed the type check?

Reply:
Apparently boolean is a subtype of int in Python, for some reason.